pkg/util/backoff/backoff.go

```diff
@@ -56,27 +62,32 @@ func NewPolicy(minBackoffFactor, baseBackoffTime, maxBackoffTime float64, recove
 		MaxBackoffTime:   maxBackoffTime,
 		RecoveryInterval: recoveryInterval,
 		MaxErrors:        maxErrors,
+		ServerlessMode:   serverlessMode,
```

Comment:
Instead of `serverlessMode`, which I think can be private here, I think we should a naming such as `constantBackoff` instead? To avoid having pkg/util/backoff/backoff.go having any serverless context.

What might be even better is to create a `Policy` interface looking like this:

```go
type Policy interface {
    GetBackoffDuration(numErrors int) time.Duration
    IncError(numErrors int) int
    DecError(numErrors int) int
}
```

The current `Policy` struct would became `ExpBackoffPolicy` and the `NewPolicy` constructor would become `NewExpBackoffPolicy` (no need of your new boolean parameter).

This done, you would be able to create the `ConstantBackoffPolicy` struct with its simple implementation just returning the duration, a `NewConstantBackoffPolicy` (with the duration as a parameter), and that would be the one you use for the Serverless Agent, while other still relies on the original implementation.

Let me know if that's not clear.

Reply:
that's a good idea! thanks! I'll update the PR

Reply:
For simplicity of reviewing I moved this to its own here: https://github.com/DataDog/datadog-agent/pull/18013